File: flutter_wycorder/lib/src/Reading.dart
// Reading.dart 
// Class for Reading

class Reading {
  int readingID;
  int system_user_id;
  DateTime time_taken;
  String status;
  int fever_chills;
  int cough;
  int sore_throat;
  int short_breath;
  int fatigue;
  int aches;
  int taste_loss;
  int congestion;
  int nausea_vomit_diarrhea;
  int infectious_contact;
  int temperature;


  // Constructor
  Reading({
    this.readingID,
    this.system_user_id,
    this.time_taken,
    this.status,
    this.fever_chills,
    this.cough, 
    this.sore_throat,
    this.short_breath,
    this.fatigue,
    this.aches,
    this.taste_loss,
    this.congestion,
    this.nausea_vomit_diarrhea,
    this.infectious_contact,
    this.temperature
  });

  // JSON Factory
  factory Reading.fromJson(Map<String, dynamic> json) {
    return Reading(
      readingID: json['readingID'],
      system_user_id: json['system_user_id'],
      time_taken: DateTime.parse(json['time_taken']),
      status: json['status'],
      fever_chills: json['fever_chills'],
      cough: json['cough'], 
      sore_throat: json['sore_throat'],
      short_breath: json['short_breath'],
      fatigue: json['fatigue'],
      aches: json['aches'],
      taste_loss: json['taste_loss'],
      congestion: json['congestion'],
      nausea_vomit_diarrhea: json['nausea_vomit_diarrhea'],
      infectious_contact: json['infectious_contact'],
      temperature: json['temperature'],
    );
  }

  Map<String, dynamic> toJson() => {
    'system_user_id':this.system_user_id.toString(),
    'time_taken':this.time_taken.toString(),
    'status':this.status,
    'fever_chills':this.fever_chills.toString(),
    'cough':this.cough.toString(),
    'sore_throat':this.sore_throat.toString(),
    'short_breath':this.short_breath.toString(),
    'fatigue':this.fatigue.toString(),
    'aches':this.aches.toString(),
    'taste_loss':this.taste_loss.toString(),
    'congestion':this.congestion.toString(),
    'nausea_vomit_diarrhea':this.nausea_vomit_diarrhea.toString(),
    'infectious_contact':this.infectious_contact.toString(),
    'temperature':this.temperature.toString()
  };


}

File: wycorder-api/api/v1/reading.py
# reading.py

# Will hold the API for the reading object


from flask import Flask, Blueprint, render_template, abort, Response, request 
from sqlalchemy.orm import sessionmaker, scoped_session, query, session 
from sqlalchemy.sql.functions import func 
import json

from models.objects import SystemUser, Reading 
from models.DB import db_session 


# Define the blueprint 
apiReading = Blueprint('apiReading', __name__, url_prefix='/api/v1/reading')

@apiReading.route('/', methods=['GET'])
def get():
    user = db_session.query(SystemUser).filter(SystemUser.token == request.headers.get('token')).one()
    if (user.system_user_id != None):
        # Token matches

        # Get the 20 most recent readings
        result = db_session.query(Reading).filter(Reading.system_user_id == user.system_user_id).order_by(Reading.time_taken.desc()).limit(20)
        
        resultJson = []
        db_session.close()

        for item in result:
            resultJson.append(item.__json__()) 

        return json.dumps(resultJson, indent=4, sort_keys=True, default=str), 200, {'content-type': 'application/json', 'Access-Control-Allow-Origin':'*'}
        
    else:
        db_session.close()
        abort(403)

@apiReading.route('/', methods=['PUT'])
def post():
    user = db_session.query(SystemUser).filter(SystemUser.token == request.headers.get('token')).one()
    if (user.system_user_id != None):
        # Token matches
        data = request.form
        reading = Reading()
        reading.system_user_id = user.system_user_id 
        reading.time_taken = data['time_taken']
        status = data['status']
        reading.fever_chills = data['fever_chills']
        reading.cough = data['cough']
        reading.sore_throat = data['sore_throat']
        reading.short_breath = data['short_breath']
        reading.fatigue = data['fatigue']
        reading.aches = data['aches']
        reading.taste_loss = data['taste_loss']
        reading.congestion = data['congestion']
        reading.nausea_vomit_diarrhea = data['nausea_vomit_diarrhea']
        reading.infectious_contact = data['infectious_contact']
        reading.temperature = data['temperature']

        db_session.add(reading)
        db_session.commit()
        db_session.close()

        return json.dumps({'status':data['status']}, indent=4, sort_keys=True, default=str), 200, {'content-type': 'application/json', 'Access-Control-Allow-Origin':'*'}
        
    else:
        db_session.close()
        abort(403)

